doc/source/security/selinux.rst

```diff
@@ -0,0 +1,507 @@
+
+        require {
+            *type mysqld_t*;
+            type var_lib_t;
+            *type default_t*;
+            class file getattr;
+            *class dir write*;
+        }
+
+        #============= mysqld_t ==============
+        *allow mysqld_t default_t:dir write*;
+        allow mysqld_t var_lib_t:file getattr;
+
+    The policy contains rules for the custom data directory and the custom logs directory. We have set the proper labels for the data directory, and applying this autogenerated policy would loosen our hardening by allowing mysqld to access ``var_lib_t`` tags. 
+
+3. Generate a policy which contains only the rules for the recent violation. Add the ``--start`` parameter to see log events after a specific time.
+
+    .. code-block:: bash
+
+        $ ausearch --start 10:00:00 -c 'mysqld' --raw | audit2allow -M my-mysqld
+
+4. Open the my_mysqld file:
+
+    .. code-block:: text
+
+        module my-mysqld 1.0;
+
+        require {
+            type mysqld_t;
+            type default_t;
+            class dir write;
+        }
+
+        #============= mysqld_t ==============
+        allow mysqld_t default_t:dir write;
+
+    This policy allows mysqld writing into the directories tagged with the default ``_t``. 
+
+5. Install the SELinux policy module:
+
+    .. code-block:: bash
+
+        $ semodule -i my-mysqld.pp
+
+After restarting the service, you have another failure. You check the journal log and view the following:
+
+.. code-block:: bash
+
+    SELinux is preventing mysqld from add_name access on the directory mysqld.log.
+
+SELinux prevents mysql from creating a log file inside the directory. 
+
+You can view all the violations by changing the SELinux mode to ``Permissive`` and then running mysqld. All violations are logged in the journal log. After this run, you can generate a local policy module, install it, and switch SELinux back to ``enforcing`` mode.
+
+1. Unload the current local my-mysqld policy module:
+
+    .. code-block:: bash
+
+        $ semodule -r my-mysqld
+
+2. You can put a single domain into permissive mode. This option allows the other domains on the system to remain in enforcing mode. Use ``semanage permissive`` with the ``-a`` parameter to change mysqld_t to permissive mode:
+
+    .. code-block:: bash
+
+        $ semanage permissive -a mysqld_t
+
+3. Verify the mode change:
+
+    .. code-block:: bash
+
+        semdule -l | grep permissive
+        ...
+        permissive_mysqld_t
+        ...
+
+4. To make searching the log easier, return the time:
+
+    .. code-block:: bash
+
+        $ date
+
+5. Start the service.
+
+    .. code-block:: bash
+
+        $ service mysqld start
+
+6. MySQL starts and SELinux logs all of the violations in the journal log. Check the journal log:
+
+    .. code-block:: bash
+
+        $ journalctl -xe
+
+7. Stop the service:
+
+    .. code-block:: bash
+
+        $ service mysqld stop 
+
+8. Generate a local mysqld policy:
```

Comment:
In place of 10:00:00 use the time got in point 4.

Reply:
Fixed